src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,348 @@
+        [Order(After = IDEDiagnosticIds.OrderModifiersDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Apply_expression_block_body_preferences))]
+        public static readonly FixIdDefinition UseExpressionBodyForIndexersDiagnosticId;
+
+        [Export]
+        [FixId(IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId)]
+        [Name(IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId)]
+        [Order(After = IDEDiagnosticIds.OrderModifiersDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Apply_expression_block_body_preferences))]
+        public static readonly FixIdDefinition UseExpressionBodyForAccessorsDiagnosticId;
+
+        [Export]
+        [FixId(IDEDiagnosticIds.InlineDeclarationDiagnosticId)]
+        [Name(IDEDiagnosticIds.InlineDeclarationDiagnosticId)]
+        [Order(After = IDEDiagnosticIds.UseImplicitTypeDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Apply_inline_out_variable_preferences))]
+        public static readonly FixIdDefinition InlineDeclarationDiagnosticId;
+
+        [Export]
+        [FixId(CSharpRemoveUnusedVariableCodeFixProvider.CS0168)]
+        [Name(CSharpRemoveUnusedVariableCodeFixProvider.CS0168)]
+        [Order(After = IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Remove_unused_variables))]
+        public static readonly FixIdDefinition CS0168;
+
+        [Export]
+        [FixId(CSharpRemoveUnusedVariableCodeFixProvider.CS0219)]
+        [Name(CSharpRemoveUnusedVariableCodeFixProvider.CS0219)]
+        [Order(After = IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Remove_unused_variables))]
+        public static readonly FixIdDefinition CS0219;
+
+        [Export]
+        [FixId(IDEDiagnosticIds.UseObjectInitializerDiagnosticId)]
+        [Name(IDEDiagnosticIds.UseObjectInitializerDiagnosticId)]
+        [Order(After = IDEDiagnosticIds.PreferBuiltInOrFrameworkTypeDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Apply_object_collection_initialization_preferences))]
+        public static readonly FixIdDefinition UseObjectInitializerDiagnosticId;
```

Comment:
I'm not a fan of this approach, as it's one more way we're introducing where parts of hte IDE have to have special knowledge about particular fixes.  This seems backwards to me (and it's what i brought up in: https://github.com/dotnet/roslyn/issues/29345).  Every time we add a feature, it seems like we're adding more and more places a feature has to have baked in knowledge about what to fixup.

This is brittle and breaks all sorts of encapsulation.  It also means we can't decouple these fixers/cleaners from the core Roslyn distro itself.  Nor can 3rd parties effectively just provide an analyzer/fixer and have all these other services light up.

It seems far more appropriate that someone should just be able to write a Analzer/Fixer and then implement/export some interfaces that end up allowing these other IDE features to discover and hook into them as appropriate.